plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/DeployJob.java

```diff
@@ -168,17 +169,19 @@ private IStatus saveCredential(Path destination) {
   }
 
   private IStatus stageProject(Path credentialFile,
-      IPath explodedWarDirectory, IPath stagingDirectory, IProgressMonitor monitor) {
-    SubMonitor progress = SubMonitor.convert(monitor, 100);
+      IPath stagingDirectory, IProgressMonitor monitor) {
     RecordProcessError stagingExitListener = new RecordProcessError();
     CloudSdk cloudSdk = getCloudSdk(credentialFile, stagingStdoutLineListener, stagingExitListener);
 
     try {
-      getJobManager().beginRule(project, progress);
-      WarPublisher.publishExploded(project, explodedWarDirectory, progress.newChild(40));
-      DeployStaging.stageStandard(explodedWarDirectory, stagingDirectory, cloudSdk,
-          progress.newChild(60));
-      return stagingExitListener.getExitStatus();
+      getJobManager().beginRule(project, null /* not worth a monitor */);
```

Comment:
The monitor is helpful in case the job is paused pending some other job that has a lock on the project.